tests/PoolTest.php

```diff
@@ -288,4 +288,29 @@ public function it_takes_an_intermediate_callback()
 
         $this->assertTrue($isIntermediateCallbackCalled);
     }
+
+    /** @test */
+    public function it_can_be_stopped_early()
+    {
+        $pool = Pool::create();
+
+        $maxProcesses = 10000;
+        $completedProcessesCount = 0;
+
+        for ($i = 0; $i < $maxProcesses; $i++) {
+            $pool->add(function () use ($i) {
+                return rand(0, 100);
```

Comment:
Let's not make this test case random. Simply return `$i`, stop when it's eg. `5`, and assert on that.

Reply:
Good idea. Submitting changes now.